tests/unit/api/test_feature.py

```diff
@@ -720,6 +720,24 @@ def test_feature__as_default_version(saved_feature):
     # check get by name and version
     assert Feature.get(name=saved_feature.name, version=new_version.version) == new_version
 
+    # check logic to prevent setting lower readiness version as default
+    saved_feature.update_readiness(FeatureReadiness.PUBLIC_DRAFT)
+    with pytest.raises(RecordUpdateException) as exc:
+        new_version.as_default_version()
+
+    error_msg = (
+        f"Cannot set default feature ID to {new_version.id} "
+        f"because it has lower readiness level than PUBLIC_DRAFT."
```

Comment:
Do we want to be more specific in the message here. e.g. "because it has lower readiness than current: PUBLIC_DRAFT."

Reply:
`PUBLIC_DRAFT` is the current max readiness, but not the current readiness.